degirum_tools/ui_support.py

```diff
@@ -193,6 +198,103 @@ class LineInfo:
     return image
 
 
+def stack_images(
+    image1: Union[np.ndarray, PIL.Image.Image],
+    image2: Union[np.ndarray, PIL.Image.Image],
+    dimension: str = 'horizontal',
+    downscale: Optional[float] = None,
+    labels: Optional[list] = None,
+    font_color: tuple = (255, 255, 255),
+) -> Union[np.ndarray, PIL.Image.Image]:
+    """ Stacks two images, either vertically or horizontally with the option two downscale
+        the images and have labels in the bottom left corner. The two images must be the same
+        size.
```

Comment:
May be it makes sense to require the same image size only in one dimension, along the gluing line? This will give more flexibility.

Reply:
@vlad-nn There are a lot of use cases where the two images are of same size (image colorization, background removal, dept estimation etc). However, there are cases where they are of different size especially use cases like super resolution where both the width and height are different.

Reply:
I'm not sure stacking for superresolution would make sense ever. What you would want to compare to is resizing to a non deep learning method, so in that case they should be the same size.

But for flexibility I now have it allowed to require only same size on the stacking dimension.